catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -91,10 +124,70 @@ public CreateRequest process(CreateRequest input)
     @Override
     public UpdateRequest process(UpdateRequest input)
             throws PluginExecutionException, StopProcessingException {
+
+        OperationTransaction operationTransaction = (OperationTransaction) input.getProperties()
+                .get(Constants.OPERATION_TRANSACTION_KEY);
+
+        if (operationTransaction == null) {
+            throw new UnsupportedOperationException(
+                    "Unable to get OperationTransaction from UpdateRequest");
+        }
+
+        if (Requests.isLocal(input)) {
+            List<Metacard> previousMetacards = operationTransaction.getPreviousStateMetacards();
+
+            boolean transientAttributeUpdates = false;
+            if (input.getProperties()
+                    .get(RegistryConstants.TRANSIENT_ATTRIBUTE_UPDATE) != null) {
+                transientAttributeUpdates = true;
+            }
+
+            Map<String, Metacard> previousMetacardsMap = new HashMap<>();
```

Comment:
Would something like this work instead of the loop?
Map<String, Metacard> previousMetacardsMap = previousMetacards.stream().collect(Collectors.toMap(Metacard::getId, Function.identity()));